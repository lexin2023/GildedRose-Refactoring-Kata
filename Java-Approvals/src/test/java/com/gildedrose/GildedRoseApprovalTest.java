package com.gildedrose;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

@UseReporter(DiffReporter.class)
public class GildedRoseApprovalTest {

	@Test
	public void ApprovalUpdateItemTest() {

        CombinationApprovals.verifyAllCombinations(
            this::doUpdateQuality,
            new String[] {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert",
                "Sulfuras, Hand of Ragnaros", "Elixir of the Mongoose", "Conjured Mana Cake"},
            new Integer[] {12, 10, 5, 0, -1},
            new Integer[] {50, 10, 5, 0, 1}
        );



	}

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose gildedRose = new GildedRose(items);
        gildedRose.updateQuality();

        return gildedRose.items[0].toString();
    }

    @Test
    public void thirtyDays() {

        ByteArrayOutputStream fakeoutput = new ByteArrayOutputStream();
        System.setOut(new PrintStream(fakeoutput));
        System.setIn(new ByteArrayInputStream("a\n".getBytes()));

        Program.main();
        String output = fakeoutput.toString();

        Approvals.verify(output);
    }
}
