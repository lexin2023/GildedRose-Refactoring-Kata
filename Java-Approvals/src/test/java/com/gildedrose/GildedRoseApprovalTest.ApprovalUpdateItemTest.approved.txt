[foo, 12, 50] => foo, 11, 49 
[foo, 12, 10] => foo, 11, 9 
[foo, 12, 5] => foo, 11, 4 
[foo, 12, 0] => foo, 11, 0 
[foo, 12, 1] => foo, 11, 0 
[foo, 10, 50] => foo, 9, 49 
[foo, 10, 10] => foo, 9, 9 
[foo, 10, 5] => foo, 9, 4 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 5, 50] => foo, 4, 49 
[foo, 5, 10] => foo, 4, 9 
[foo, 5, 5] => foo, 4, 4 
[foo, 5, 0] => foo, 4, 0 
[foo, 5, 1] => foo, 4, 0 
[foo, 0, 50] => foo, -1, 48 
[foo, 0, 10] => foo, -1, 8 
[foo, 0, 5] => foo, -1, 3 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, -1, 50] => foo, -2, 48 
[foo, -1, 10] => foo, -2, 8 
[foo, -1, 5] => foo, -2, 3 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[Aged Brie, 12, 50] => Aged Brie, 11, 50 
[Aged Brie, 12, 10] => Aged Brie, 11, 11 
[Aged Brie, 12, 5] => Aged Brie, 11, 6 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 1] => Aged Brie, 11, 2 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 10, 10] => Aged Brie, 9, 11 
[Aged Brie, 10, 5] => Aged Brie, 9, 6 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 5, 10] => Aged Brie, 4, 11 
[Aged Brie, 5, 5] => Aged Brie, 4, 6 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 5] => Aged Brie, -1, 7 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 5] => Aged Brie, -2, 7 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 10] => Backstage passes to a TAFKAL80ETC concert, 11, 11 
[Backstage passes to a TAFKAL80ETC concert, 12, 5] => Backstage passes to a TAFKAL80ETC concert, 11, 6 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert, 9, 12 
[Backstage passes to a TAFKAL80ETC concert, 10, 5] => Backstage passes to a TAFKAL80ETC concert, 9, 7 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 10] => Backstage passes to a TAFKAL80ETC concert, 4, 13 
[Backstage passes to a TAFKAL80ETC concert, 5, 5] => Backstage passes to a TAFKAL80ETC concert, 4, 8 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 5] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50 
[Sulfuras, Hand of Ragnaros, 12, 10] => Sulfuras, Hand of Ragnaros, 12, 10 
[Sulfuras, Hand of Ragnaros, 12, 5] => Sulfuras, Hand of Ragnaros, 12, 5 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 1] => Sulfuras, Hand of Ragnaros, 12, 1 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros, 10, 10 
[Sulfuras, Hand of Ragnaros, 10, 5] => Sulfuras, Hand of Ragnaros, 10, 5 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 10] => Sulfuras, Hand of Ragnaros, 5, 10 
[Sulfuras, Hand of Ragnaros, 5, 5] => Sulfuras, Hand of Ragnaros, 5, 5 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 5] => Sulfuras, Hand of Ragnaros, -1, 5 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Elixir of the Mongoose, 12, 50] => Elixir of the Mongoose, 11, 49 
[Elixir of the Mongoose, 12, 10] => Elixir of the Mongoose, 11, 9 
[Elixir of the Mongoose, 12, 5] => Elixir of the Mongoose, 11, 4 
[Elixir of the Mongoose, 12, 0] => Elixir of the Mongoose, 11, 0 
[Elixir of the Mongoose, 12, 1] => Elixir of the Mongoose, 11, 0 
[Elixir of the Mongoose, 10, 50] => Elixir of the Mongoose, 9, 49 
[Elixir of the Mongoose, 10, 10] => Elixir of the Mongoose, 9, 9 
[Elixir of the Mongoose, 10, 5] => Elixir of the Mongoose, 9, 4 
[Elixir of the Mongoose, 10, 0] => Elixir of the Mongoose, 9, 0 
[Elixir of the Mongoose, 10, 1] => Elixir of the Mongoose, 9, 0 
[Elixir of the Mongoose, 5, 50] => Elixir of the Mongoose, 4, 49 
[Elixir of the Mongoose, 5, 10] => Elixir of the Mongoose, 4, 9 
[Elixir of the Mongoose, 5, 5] => Elixir of the Mongoose, 4, 4 
[Elixir of the Mongoose, 5, 0] => Elixir of the Mongoose, 4, 0 
[Elixir of the Mongoose, 5, 1] => Elixir of the Mongoose, 4, 0 
[Elixir of the Mongoose, 0, 50] => Elixir of the Mongoose, -1, 48 
[Elixir of the Mongoose, 0, 10] => Elixir of the Mongoose, -1, 8 
[Elixir of the Mongoose, 0, 5] => Elixir of the Mongoose, -1, 3 
[Elixir of the Mongoose, 0, 0] => Elixir of the Mongoose, -1, 0 
[Elixir of the Mongoose, 0, 1] => Elixir of the Mongoose, -1, 0 
[Elixir of the Mongoose, -1, 50] => Elixir of the Mongoose, -2, 48 
[Elixir of the Mongoose, -1, 10] => Elixir of the Mongoose, -2, 8 
[Elixir of the Mongoose, -1, 5] => Elixir of the Mongoose, -2, 3 
[Elixir of the Mongoose, -1, 0] => Elixir of the Mongoose, -2, 0 
[Elixir of the Mongoose, -1, 1] => Elixir of the Mongoose, -2, 0 
[Conjured Mana Cake, 12, 50] => Conjured Mana Cake, 11, 48 
[Conjured Mana Cake, 12, 10] => Conjured Mana Cake, 11, 8 
[Conjured Mana Cake, 12, 5] => Conjured Mana Cake, 11, 3 
[Conjured Mana Cake, 12, 0] => Conjured Mana Cake, 11, 0 
[Conjured Mana Cake, 12, 1] => Conjured Mana Cake, 11, 0 
[Conjured Mana Cake, 10, 50] => Conjured Mana Cake, 9, 48 
[Conjured Mana Cake, 10, 10] => Conjured Mana Cake, 9, 8 
[Conjured Mana Cake, 10, 5] => Conjured Mana Cake, 9, 3 
[Conjured Mana Cake, 10, 0] => Conjured Mana Cake, 9, 0 
[Conjured Mana Cake, 10, 1] => Conjured Mana Cake, 9, 0 
[Conjured Mana Cake, 5, 50] => Conjured Mana Cake, 4, 48 
[Conjured Mana Cake, 5, 10] => Conjured Mana Cake, 4, 8 
[Conjured Mana Cake, 5, 5] => Conjured Mana Cake, 4, 3 
[Conjured Mana Cake, 5, 0] => Conjured Mana Cake, 4, 0 
[Conjured Mana Cake, 5, 1] => Conjured Mana Cake, 4, 0 
[Conjured Mana Cake, 0, 50] => Conjured Mana Cake, -1, 46 
[Conjured Mana Cake, 0, 10] => Conjured Mana Cake, -1, 6 
[Conjured Mana Cake, 0, 5] => Conjured Mana Cake, -1, 1 
[Conjured Mana Cake, 0, 0] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 1] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, -1, 50] => Conjured Mana Cake, -2, 46 
[Conjured Mana Cake, -1, 10] => Conjured Mana Cake, -2, 6 
[Conjured Mana Cake, -1, 5] => Conjured Mana Cake, -2, 1 
[Conjured Mana Cake, -1, 0] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 1] => Conjured Mana Cake, -2, 0 
